src/redux/actions/authActions.js

```diff
@@ -0,0 +1,54 @@
+import {
+  SIGNUP_SUCCESS,
+  SIGNUP_ERROR,
+  SIGNUP_INPUT,
+  UPDATE_SPINNER_STATUS,
+} from '../actionTypes/authActionTypes';
+import apiCall from '../../helpers/apiCall';
+import {
+  notificationError,
+  notificationSuccess,
+} from '../../helpers/notificationPopUp';
+
+export const updateSignupInputAction = (data) => ({
+  type: SIGNUP_INPUT,
+  payload: data,
+});
+export const spinnerStatusAction = () => ({
+  type: UPDATE_SPINNER_STATUS,
+  payload: {},
+});
+
+export const signupAction = (signupData) => (signupData.password !== signupData.confirmPassword
+  ? async (dispatch) => {
+    notificationError('The password and confirm password should match');
+    dispatch({
+      type: SIGNUP_ERROR,
+      payload: {
+        message: 'The password and confirm password should match',
+      },
+    });
+  }
+  : async (dispatch) => {
+    dispatch(spinnerStatusAction());
+    try {
+      const signupResponse = await apiCall.post('/auth/signup', signupData);
+      notificationSuccess(signupResponse.data.message);
+      localStorage.setItem('token', signupResponse.data.data);
+      dispatch({
+        type: SIGNUP_SUCCESS,
+        payload: signupResponse.data,
+      });
+    } catch (error) {
+      let err = error.request ? error.response = { data: { message: 'Network Error' } } : error.response;
+      err = error.response.data.message
+        ? error.response.data.message
+        : error.response.data.error;
+      err = typeof err === 'object' ? err : [err];
+      err.map((e) => notificationError(e));
+      dispatch({
+        type: SIGNUP_ERROR,
```

Comment:
@hezronkimutai can you work on this try/catch block?
The feature is working properly when I provide incomplete/wrong info. However, When I provide correct information, I'm getting `network error`.